cli/listener/src/main.rs

```diff
@@ -220,35 +218,309 @@ pub fn insert_price(conn: &Connection, exchange_rate: ExchangeRate) -> rusqlite:
     Ok(())
 }
 
+struct Daemon<'a, 'b> {
+    config: &'a mut SnapshotClientConfig<'b>,
+    opts: &'a Opts,
+
+    /// Mutex where we publish the latest snapshot for use by the webserver.
+    snapshot_mutex: Arc<SnapshotMutex>,
```

Comment:
So for `solido run-maintainer`, we keep the latest snapshot of the state in memory in an Arc, so we can serve the metrics from it. But for this daemon, we plan to instead read from the state and insert a row into the database based on that, and for any endpoints, we will read from the database. So we could keep the `Metrics` in an Arc, but all of the on-chain data should go in the database instead; we don’t need this snapshot here.